### ISSUE TITLE
[6.x] Pet System

### ISSUE THREAD

**Issue by nerv234** [TrinityCore/TrinityCore#18024]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description**:

With Legion there are no spell auras that controlls stats of pets. Only one spell that remains is:
ID - 51906 Death Knight Rune Weapon Scaling

So with this there should be some source of information that indicates how system should work.
Maybe this isn't complete report but i started it to start discusion about pet system in legion.

Btw. sorry for offtopic, but is there any tool to view content of db2 files?

**Branch(es)**: 3.3.5 / 6.x (Select the branch(es) affected by this issue) 

**TC hash/commit**:  https://github.com/TrinityCore/TrinityCore/commit/e2c92dbc63a16f07bd46b2423cfe23fc22ab90b0

**TDB version**:  last

**Operating system**:  win7 (not related)

**Comment by jackpoz** [comment]
> TC hash/commit: last

you ignored the issue template that requires to specify the commit hash (and also explains that there's no "latest" commit)

**Comment by nerv234** [comment]
Updated hash.

**Comment by matukaa** [comment]
> Btw. sorry for offtopic, but is there any tool to view content of db2 files?

https://github.com/Shauren/dbcviewer

**Comment by MrSmite** [comment]
> With Legion there are no spell auras that controlls stats of pets.

Not sure what you mean by this but pet specializations are still done using auras:

> [Great Stamina](http://www.wowhead.com/spell=61688/great-stamina)
> Apply Aura: Mod Increase Maximum Health - % (1)
Value: 60%

**Comment by nerv234** [comment]
What about strength/spell power scaling, haste scaling with owner etc.?

**Comment by funjoker** [comment]
In Legion:

Main Pet: 
- 60% AP
- 60% AP -> SP
- 170% Armor
- 70% Health
- 125% Focus Regen

Hati & Dire Best & Black Arrow:
- 115% AP
- 20% AP -> SP
- 125% Armor
- 20% Health

**Comment by funjoker** [comment]
Warlock Pets:

All Pets:
- 75% SP

SP -> AP varies by Pet (Formula: SP * Factor = PetAP)
- Abyssal: 0.04688
- Doomguard: 0.09375
- Infernal: 0.04688
- Terrorguard: 0.09375
- Wild Imp: 0.09375
- Fel Imp: 0.6563
- Felguard: 0.6563
- Felhunter: 0.6563
- Imp: 0.6563
- Oberserver: 0.6563
- Shivarra: 0.2081
- Succubus: 0.3124
- Voidlord: 0.6563
- Voidwalker: 0.6563
- Wrathguard: 0.4376

**Comment by funjoker** [comment]
Probably Pet Auto-Attack of BM Hunter:

HunterAP * 0.333 / 3.5 * PetAttackSpeed * 1.5 Combat Expertise (1.85 when Adaptation) * 1.2 Empowered Pets (when Empowered Pets) * (1 + Mastery%/100) = Auto Attack Dmg PREARMOR

**Comment by DoctorKraft** [comment]
Exactly, the better solution is delete old table "pet_levelstats". Add new with data for armor, sp/ap, health, powertype, attackspeed, etc...

Also pet stable, call pet button (warlock/hunter) and pet spec for hunter is not updating

**Comment by funjoker** [comment]
I guess no pet has any specific data anymore.
All pet stats are depending on hunter itself.

**Comment by funjoker** [comment]
Gonna start work on this :/ super annoying that it's so outdated

**Comment by funjoker** [comment]
https://gist.github.com/funjoker/e7108c1a081452e3cec6b8a8ad7203c8

This is a list of all CreatureIds used for Guardian/Pet summons.

**Comment by funjoker** [comment]
do minions receive helath or anything ?

**Comment by funjoker** [comment]
Health% of Owner 
Imp: 30%
Voidwalker:  50%
Felhunter: 40%
Succubus: 40%
Felguard: 50%
Dreadstalker: 40%
Wild Imp: 8010 15%
Doomguard: 40%
Infernal: 50%
Darkglare: 40%
